app.py

```diff
@@ -1,247 +1,64 @@
-                           state=state)
-    dest = furl(get_openid_config(openid_provider)["authorization_endpoint"]).add(query_params=auth_params).url
-    return Response(status_code=302, headers=dict(Location=dest), body="")
-
-
-@app.route('/.well-known/openid-configuration')  # Part of OIDC
-def serve_openid_config():
-    openid_config = get_openid_config(os.environ["OPENID_PROVIDER"])
-    auth_host = app.current_request.headers['host']
-    openid_config.update(authorization_endpoint=f"https://{auth_host}/authorize",
-                         token_endpoint=f"https://{auth_host}/oauth/token",
-                         jwks_uri=f"https://{auth_host}/.well-known/jwks.json",
-                         revocation_endpoint=f"https://{auth_host}/oauth/revoke",
-                         userinfo_endpoint=f"https://{auth_host}/userinfo")
-    return Response(status_code=200, body=openid_config)
-
-
-def proxy_response(dest_url, **extra_query_params):
-    query_params = app.current_request.query_params or {}
-    if query_params or extra_query_params:
-        dest_url = furl(dest_url).add(dict(query_params, **extra_query_params)).url
-    proxy_res = requests.request(method=app.current_request.method,
-                                 url=dest_url,
-                                 headers=app.current_request.headers,
-                                 data=app.current_request.raw_body)
-    if proxy_res.headers.get("Content-Type", "").startswith("application/json"):
-        body = proxy_res.json()
-    else:
-        body = proxy_res.content.decode()
-    for header in "connection", "content-length", "date":
-        proxy_res.headers.pop(header, None)
-    return Response(status_code=proxy_res.status_code,
-                    headers=dict(proxy_res.headers),
-                    body=body)
-
-
-@app.route('/.well-known/jwks.json')  # Part of OIDC
-def serve_jwks_json():
-    openid_config = get_openid_config(os.environ["OPENID_PROVIDER"])
-    return proxy_response(openid_config["jwks_uri"])
-
-
-@app.route('/oauth/revoke')  # Part of OIDC
-def revoke():
-    openid_config = get_openid_config(os.environ["OPENID_PROVIDER"])
-    return proxy_response(openid_config["revocation_endpoint"])
-
-
-@app.route('/userinfo')  # Part of OIDC
-def userinfo():
-    openid_config = get_openid_config(os.environ["OPENID_PROVIDER"])
-    return proxy_response(openid_config["userinfo_endpoint"])
-
-
-@app.route('/oauth/token', methods=["POST"], content_types=["application/x-www-form-urlencoded",
-                                                            "application/x-www-form-urlencoded;charset=UTF-8"])
-def serve_oauth_token():  # Part of OIDC
-    # TODO: client id/secret mgmt
-    openid_provider = os.environ["OPENID_PROVIDER"]
-    openid_config = get_openid_config(openid_provider)
-    return proxy_response(openid_config["token_endpoint"])
+@app.route('/swagger.json')
```

Comment:
This and the Swagger UI should be available automatically from connexion (see e.g. https://query.dev.data.humancellatlas.org/v1/openapi.json, https://query.dev.data.humancellatlas.org/v1/ui/)

Reply:
it has been proven to more difficult than I thought to get this work. Creating a ticket to fix later https://github.com/HumanCellAtlas/fusillade/issues/100.